### PR TITLE
Fix(Script/Quest) Abduction

### PR DESCRIPTION
**Pull request by ghost**
**Changes proposed:**

-  Change scripting to handle quest more blizzlike.

**Tests performed:**

SQL inserts, Scripts compile, Tested quest in game.

To Test:

.tele Borean
. quest add 11590
Fight a Beryl Sorcerer until he is below 25% health.
Use Arcane Binder item on Beryl Sorcerer.
Return to Librarian Donathan.

Note: I am rearranging the scripts by order of quest number.

### CODE REVIEW

File: src/server/scripts/Northrend/zone_borean_tundra.cpp [hunk 1]
```diff
@@ -34,6 +34,141 @@
...
+    void SpellHit(WorldObject* caster, SpellInfo const* spellInfo) override
+    {
+        if (_chainsCast)
+            return;
+
+        if (spellInfo->Id == SPELL_ARCANE_CHAINS)
+            if (Player* player = caster->ToPlayer())
+            {
+                _playerGUID = player->GetGUID();
+                _chainsCast = true;
```
Comment (jackpoz) [inline src/server/scripts/Northrend/zone_borean_tundra.cpp:85]: is this supposed to be set to false somewhere again ? like on Reset() ?

Comment (malcrom) [reply src/server/scripts/Northrend/zone_borean_tundra.cpp:85]: He despawns in script

Comment (malcrom) [reply src/server/scripts/Northrend/zone_borean_tundra.cpp:85]: false is set on spawn

File: sql/updates/world/3.3.5/Abduction.sql [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+INSERT INTO `conditions` (`SourceTypeOrReferenceId`, `SourceGroup`, `SourceEntry`, `SourceId`, `ElseGroup`, `ConditionTypeOrReference`, `ConditionTarget`, `ConditionValue1`, `ConditionValue2`, `ConditionValue3`, `NegativeCondition`, `ErrorType`, `ErrorTextId`, `ScriptName`, `Comment`) VALUES
+(17, 0, 45611, 0, 0, 31, 1, 3, 25316, 0, 0, 12, 0, '', 'Spell Arcane Chains will hit the explicit target of the spell if target is unit Beryl Sorcerer.'),
+(17, 0, 45611, 0, 0, 38, 1, 25, 4, 0, 0, 12, 0, '', 'Spell Arcane Chains will hit the explicit target of the spell if target health percentage must be equal or lower than 25% of max Health.');
+
+DELETE FROM `spell_script_names` WHERE `spell_id`= 45625;
+INSERT INTO `spell_script_names` (`spell_id`,`ScriptName`) VALUES
+(45625, 'spell_arcane_chains_character_force_cast');
+
+-- Update creatures
+UPDATE `creature_template` SET `minlevel`=69, `maxlevel`=70, `unit_flags`=32768, `unit_class`=8, `ScriptName`='npc_captured_beryl_sorcerer' WHERE `entry`=25474;
```
Comment (offl) [inline sql/updates/world/3.3.5/Abduction.sql:14]: He takes(or should take) level from player, in DB levels are correct
```txt
    Id  CreatureId  FactionTemplateId  Expansion  MinLevel  MaxLevel     Flags1  Flags2  Flags3  Flags4  Flags5  
------  ----------  -----------------  ---------  --------  --------  ---------  ------  ------  ------  --------
 24118       25474                 35          2        71        72  268959750       0       0       0         0

Comment (malcrom) [reply sql/updates/world/3.3.5/Abduction.sql:14]: [2] Level: 70 -- from sniff

Comment (malcrom) [reply sql/updates/world/3.3.5/Abduction.sql:14]: [2] ClassId: 2 -- from sniff so I got that wrong.

Comment (malcrom) [reply sql/updates/world/3.3.5/Abduction.sql:14]: So I should inform you that, unless they have been updated, all exp, class, and level values where put there by me an long while back over a week of 6 hours a day gathering data from mostly wowhead. So never assume that they are all 100% especially npcs that are summoned for a quest.

**Comment by malcrom** [comment]
While waiting for approval: https://www.youtube.com/watch?v=S0L_n68bznE

File: sql/updates/world/3.3.5/Abduction.sql (hunk 2 above)
Comment (offl) [reply sql/updates/world/3.3.5/Abduction.sql]: This creature takes level from player after summon, so level in sniff or wowhead is wrong, SQL row in my comment above is basically a piece of Blizz DB, same values are used currently in TC and doesn't require changes

File: src/server/scripts/Northrend/zone_borean_tundra.cpp [hunk 3]
```diff
@@ -34,6 +34,138 @@
...
+
+// NPC 25474: Captured Beryl Sorcerer
+struct npc_captured_beryl_sorcerer : public FollowerAI
+{
+    npc_captured_beryl_sorcerer(Creature* creature) : FollowerAI(creature) { }
+
+    void JustAppeared() override
+    {
+        me->SetReactState(REACT_DEFENSIVE);
+        me->SetFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_NON_ATTACKABLE);
```
Comment (offl) [inline src/server/scripts/Northrend/zone_borean_tundra.cpp:133]: SetImmuneToAll please, without this flag
I know it doesn't have any of those flags but if hacking something, better to do it without that flag

File: sql/updates/world/3.3.5/Abduction.sql (hunk 2 above)
Comment (malcrom) [reply sql/updates/world/3.3.5/Abduction.sql]: I was 71 when I did this quest and the npc was 70.

Comment (offl) [reply sql/updates/world/3.3.5/Abduction.sql]: 80 and 80 in 335 sniff
Checked a video from MoP, he changes level to player's (90 > 90). That is weird they changed level in DB at some point, maybe even in WotLK. Better to keep old values to reduce mismatches with db2

File: src/server/scripts/Northrend/zone_borean_tundra.cpp (hunk 3 above)
Comment (malcrom) [reply src/server/scripts/Northrend/zone_borean_tundra.cpp]: done

**Comment by malcrom** [comment]
Hey Kevin! Can you accept a fix?

**Comment by malcrom** [comment]
Hey Kevin! Can you accept a fix?

**Pull request merged**